etna/models/nn/patchts.py

```diff
@@ -0,0 +1,379 @@
+            .drop(["target"], axis=1)
+            .pipe(lambda x: x[["target_shifted"] + [i for i in x.columns if i != "target_shifted"]])
+            .values
+        )
+        values_target = df["target"].values
+        segment = df["segment"].values[0]
+
+        def _make(
+                values_real: np.ndarray,
+                values_target: np.ndarray,
+                segment: str,
+                start_idx: int,
+                encoder_length: int,
+                decoder_length: int,
+        ) -> Optional[dict]:
+
+            sample: Dict[str, Any] = {
+                "encoder_real": list(),
+                "decoder_real": list(),
+                "encoder_target": list(),
+                "decoder_target": list(),
+                "segment": None,
+            }
+            total_length = len(values_target)
+            total_sample_length = encoder_length + decoder_length
+
+            if total_sample_length + start_idx > total_length:
+                return None
+
+            # Get shifted target and concatenate it with real values features
+            sample["decoder_real"] = values_real[start_idx + encoder_length: start_idx + total_sample_length]
+
+            # Get shifted target and concatenate it with real values features
+            sample["encoder_real"] = values_real[start_idx: start_idx + encoder_length]
+            sample["encoder_real"] = sample["encoder_real"][1:]
+
+            target = values_target[start_idx: start_idx + encoder_length + decoder_length].reshape(-1, 1)
+            sample["encoder_target"] = target[1:encoder_length]
+            sample["decoder_target"] = target[encoder_length:]
+
+            sample["segment"] = segment
+
+            return sample
+
+        start_idx = 0
+        while True:
+            batch = _make(
+                values_target=values_target,
+                values_real=values_real,
+                segment=segment,
+                start_idx=start_idx,
+                encoder_length=encoder_length,
+                decoder_length=decoder_length,
+            )
+            if batch is None:
+                break
+            yield batch
+            start_idx += 1
+
+    def configure_optimizers(self) -> "torch.optim.Optimizer":
+        """Optimizer configuration."""
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.lr, **self.optimizer_params)
+        return optimizer
+
+
+class PatchTSModel(DeepBaseModel):
+    """PatchTS model using PyTorch layers"""
+
+    def __init__(
+            self,
+            decoder_length: int,
+            encoder_length: int,
+            patch_len: int = 4,
+            stride: int = 1,
+            num_layers: int = 3,
+            hidden_size: int = 128,
+            feedforward_size: int = 256,
+            nhead: int = 16,
+            lr: float = 1e-3,
+            loss: Optional["torch.nn.Module"] = None,
+            train_batch_size: int = 16,
```

Comment:
If it's ok to make batch larger - we wouldn't have problems with memory, let's make  it